syntax = "proto3";

package menu;

service MenuService {
    rpc AddMenuItem(MenuItemRequest) returns (MenuItemResponse);
    rpc GetMenuItems(Empty) returns (MenuItemsResponse);
    rpc UpdateMenuItem(MenuItemRequest) returns (MenuItemResponse);
    rpc DeleteMenuItem(MenuItemRequest) returns (DeleteResponse);
}

message Empty {}

message MenuItemRequest {
    int32 id = 1;
    string name = 2;
    string category = 3;
    double price = 4;
}

message MenuItemResponse {
    int32 id = 1;
    string name = 2;
    string category = 3;
    double price = 4;
}

message MenuItemsResponse {
    repeated MenuItemResponse items = 1;
}

message DeleteResponse {
    bool success = 1;
}
